dotnet/src/SemanticKernel/SkillDefinition/FunctionBase.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.AI.TextCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+using Microsoft.SemanticKernel.Events;
+using Microsoft.SemanticKernel.Orchestration;
+
+namespace Microsoft.SemanticKernel.SkillDefinition;
+internal abstract class FunctionBase : ISKFunction,
```

Comment:
Can you add some documentation on this class and its purpose? Why does a Plan not implement FunctionBase?

Reply:
Using the `FunctionBase` was not initially in the scope of this PR but I did it due to the simplicity + benefits except for Plans:

1. `FunctionBase` is intended to be internal while `Plan` is public.
2. `FunctionBase` implements IList<ParameterView> in Parameters property while `Plan` has `ContextVariables` as Parameters property
3. Doing that for plans in this PR brings more complexity to the actual Plan implementation (many different contructors that would need to the current FunctionBase abstraction.